src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +34,12 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .map(index -> !isEven(index) ? numbers.get(index) - 1 : numbers.get(index))
+                .filter(n -> !isEven(n))
```

Comment:
! is rather hard to be identified during expressions reading, better to avoid it if possible
```suggestion
                .map(index -> isOdd(index) ? numbers.get(index) - 1 : numbers.get(index))
                .filter(n -> isOdd(n))
```
and add method
```
        private boolean isOdd(int n) {
            return !isEven(n);
        }
```